spec/shortcuts_spec.rb

```diff
@@ -159,6 +159,40 @@
     end
   end
   
+  context "::magic" do
+  
+    [:dense, :list, :yale].each do |stype|
+      context stype do
+ 	
+        ALL_DTYPES.each do |dtype|
+          context dtype do  
+            it "creates a matrix with numbers from 1 to n^n(n squared)" do
+              a = NMatrix.magic(3)
+              magic3 = NMatrix.new([3,3], [4, 9, 2, 3, 5, 7, 8, 1, 6], dtype: dtype, stype: stype)	
```

Comment:
The reason why I'm saying so is because there is NOWHERE in the `.magic` function where you specify the dtype or the stype, so _those loops are not necessary_.